File: flask-madlibs/stories.py
"""Madlibs Stories."""


class Story:
    """Madlibs story.

    To  make a story, pass a list of prompts, and the text
    of the template.

        >>> s = Story
        ...     ["simple",
        ...     "A Simple Tale",
        ...     ["noun", "verb"],
        ...     "I love to {verb} a good {noun}.")

    To generate text from a story, pass in a dictionary-like thing
    of {prompt: answer, promp:answer):

        >>> ans = {"verb": "eat", "noun": "mango"}
        >>> s.generate(ans)
        'I love to eat a good mango.'
    """

    def __init__(self, code, title, words, text):
        """Create story with words and template text."""

        self.code = code
        self.title = title
        self.prompts = words
        self.template = text

    def generate(self, answers):
        """Substitute answers into text."""

        text = self.template

        for (key, val) in answers.items():
            text = text.replace("{" + key + "}", val)

        return text


# Here's a story to get you started


story1 = Story(
    "history",
    "A History Tale",
    ["place", "noun", "verb", "adjective", "plural_noun"],
    """Once upon a time in a long-ago {place}, there lived a
       large {adjective} {noun}. It loved to {verb} {plural_noun}."""
)

story2 = Story(
    "adventure",
    "An Adventure Story",
    ["place", "noun", "verb", "adjective", "plural_noun", "noun2"],
    """One {adjective} day, a {noun} took a trip to {place}. It saw three {plural_noun} {verb} a {noun2}."""
)

# Make a dictionary of all stories, so we can look them up by code.
stories = {s.code: s for s in [story1, story2]}

File: react-1/madlibs-app/src/stories.js
const stories = [
    { id: 'story1', 
    name: 'History', 
    prompts: ['place', 'noun', 'adjective', 'verbPast', 'color', 'pluralNoun'],
    template: 'Once upon a time in a long-ago {place}, there lived a large {adjective} {noun} that {verbPast} {color} {pluralNoun}.', 
    },
    { id: 'story2', 
    name: 'Scary', 
    prompts: ['place', 'noun', 'adjective', 'verbPast', 'color', 'pluralNoun'],
    template: 'One dark night in {place}, a {adjective} {noun} crept up along side some {color} {pluralNoun} and {verbPast} them.',
    },
    { id: 'story3',
    name: 'Romance',
    prompts: ['place', 'noun', 'adjective', 'verbPast', 'color', 'pluralNoun'],
    template: 'A long time ago in {place}, a {adjective} man gave a young girl two {color} {pluralNoun} and {verbPast} her under a {noun}.'
    },
    { id: 'story4',
    name: 'Comedy',
    prompts: ['place', 'noun', 'adjective', 'verbPast', 'color', 'pluralNoun'],
    template: 'There once was a {adjective} {noun} that {verbPast} three {color} {pluralNoun} in {place}.',
    },
    // Add more story templates as needed
  ];

export default stories;